### PR TITLE
Support sparse for some operators in mt.linalg

### LINKED ISSUE #211: Support sparse for linalg.solve and linalg.lu
**Is your feature request related to a problem? Please describe.**

Support some linalg operands for sparse tensors, such as `mt.linalg.inv` `mt.linalg.lu`.

We should support this.
``` Python
import scipy.sparse as sps
import mars.tensor as mt

data = sps.random(9, 9, density=.1)
arr = mt.tensor(data, chunk_size=3)
arr_inv = mt.linalg.inv(arr)
```

**Describe the solution you'd like**
Add supported sparse linalg function in class `SparseMatrix`.

### PR DESCRIPTION
**Pull request by NJUPole**
## What do these changes do?

Support sparse for `mt.linalg.lu`, `mt.linalg.solve_triangular`, `mt.linalg.solve`, `mt.linalg.inv`.

## Related issue number

resolve #211

### CODE REVIEW

**Comment by chineking** [comment]
I think this feature can be backported.

**Comment by wjsi** [comment]
Take a look at duplication data in https://app.codacy.com/app/mars-project/mars/pullRequest?prid=3105813, I think at least there is sth can be done.

File: mars/tensor/execution/linalg.py [hunk 1]
```diff
@@ -68,22 +68,30 @@ def _solve_triangular(ctx, chunk):
...
         [ctx[c.key] for c in chunk.inputs], device=chunk.device, ret_extra=True)
 
     with device(device_id):
-        p, l, u = scipy.linalg.lu(a)
+        if chunk.is_sparse():
+            from .array import sparse
+
+            p, l, u = sparse.lu(a)
+        else:
+            import scipy.linalg
```
Comment (chineking) [inline mars/tensor/execution/linalg.py]: Ditto.

File: mars/tensor/execution/linalg.py [hunk 2]
```diff
@@ -72,24 +72,25 @@ def _solve_triangular(ctx, chunk):
             import scipy.linalg
 
             ctx[chunk.key] = scipy.linalg.solve_triangular(a, b, lower=chunk.op.lower)
-        else:
+        elif is_sparse_module(xp):
```
Comment (chineking) [inline mars/tensor/execution/linalg.py]: Cupy now supports solve_triangular, refer to https://docs-cupy.chainer.org/en/stable/reference/generated/cupyx.scipy.linalg.solve_triangular.html#cupyx.scipy.linalg.solve_triangular

No else any more.

File: mars/tensor/expressions/linalg/solve.py [hunk 3]
```diff
@@ -63,6 +63,6 @@ def solve(a, b, sym_pos=False):
         u = l.T
     else:
         p, l, u = lu(a)
-        b = p.T.dot(b, sparse=a.issparse())
+        b = p.T.dot(b, sparse=False)
```
Comment (chineking) [inline mars/tensor/expressions/linalg/solve.py]: I think `b` here can be sparse.

**Comment by chineking** [comment]
I leave some comments, also check the code coverage and fix the issue raised by Codacy.

File: mars/tensor/execution/linalg.py [hunk 4]
```diff
@@ -72,10 +72,8 @@ def _solve_triangular(ctx, chunk):
             import scipy.linalg
 
             ctx[chunk.key] = scipy.linalg.solve_triangular(a, b, lower=chunk.op.lower)
-        elif is_sparse_module(xp):
-            ctx[chunk.key] = xp.solve_triangular(a, b, lower=chunk.op.lower)
         else:
-            raise NotImplementedError
```
Comment (chineking) [inline mars/tensor/execution/linalg.py]: add cupy support here:

```python
elif xp is cp:
    import cupyx.scipy.linalg 
   ctx[chunk.key] = cupyx.scipy.linalg.solve_triangular(a, b, lower=chunk.op.lower)
```

**Pull request merged**